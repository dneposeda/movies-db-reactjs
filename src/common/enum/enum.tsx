export enum TypeNameBtnGroups {
    SORT = 'sort',
    SEARCH = 'search',
}

export enum TypeSearch {
    TITLE = 'title',
    GENRE = 'genres',
}

export enum SearchButtonNames {
    TITLE = 'title',
    GENRE = 'genre',
}

export enum TypeSort {
    DATE = 'release_date',
    RATING = 'vote_average',
}

export enum SortButtonNames {
    DATE = 'release date',
    RATING = 'rating'
}
