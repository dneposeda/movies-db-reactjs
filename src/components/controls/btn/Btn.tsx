import React, {FC} from 'react';
import classNames from 'classnames';

interface ButtonProps {
    title: string,
    cls?: string[],
    active?: boolean,
    onClick?: () => void,
}
type Component = FC<ButtonProps>;

const Btn: Component = ({ title , cls, active, onClick}) => (
    <button
        type="button"
        className={ classNames(cls, { active: active }) }
        onClick={ onClick }
    >
        { title }
    </button>
)

export default Btn;
