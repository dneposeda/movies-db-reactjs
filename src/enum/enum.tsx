enum TypeNameBtnGroups {
    SORT = 'sort',
    SEARCH = 'search',
}

enum TypeSearch {
    TITLE = 'title',
    GENRE = 'genre',
}

enum TypeSort {
    DATE = 'release date',
    RATING = 'rating',
}

export { TypeNameBtnGroups, TypeSearch, TypeSort };
