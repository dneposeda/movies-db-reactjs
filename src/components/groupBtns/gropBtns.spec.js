import React from 'react';
import GropBtns from './GropsBtns';
import { shallow } from 'enzyme';
import Btn from '../controls/btn/Btn';
import { TypeNameBtnGroups, TypeSort } from '../../enum/enum'

const mockBtns = {
    btns: [
        { id: 1, title: TypeSort.DATE, active: true },
        { id: 2, title: TypeSort.RATING, active: false },
    ],
    classesBtn: ['btn', 'filter-sort__btn'],
    titleGroup: TypeNameBtnGroups.SORT,
    handleClickBtn: () => {},
}

describe('GropBtns', () => {
    it('renders correctly', () => {
        const component = mount (
            <GropBtns
                btns={ mockBtns.btns }
                cls={ mockBtns.classesBtn }
                group={ mockBtns.titleGroup }
                handleClickBtn={ mockBtns.handleClickBtn }
            />
        );

        expect(component).toMatchSnapshot();
        expect(component.find(Btn)).toHaveLength(2);
    });
});
