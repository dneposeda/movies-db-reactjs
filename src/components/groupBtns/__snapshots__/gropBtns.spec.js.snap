// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GropBtns renders correctly 1`] = `
<div
  className="btn-group"
  role="group"
>
  <Btn
    active={true}
    cls={
      Array [
        "btn",
        "filter-sort__btn",
      ]
    }
    handleClick={[Function]}
    key="1"
    title="release date"
  />
  <Btn
    active={false}
    cls={
      Array [
        "btn",
        "filter-sort__btn",
      ]
    }
    handleClick={[Function]}
    key="2"
    title="rating"
  />
</div>
`;
